File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-2_Gestion-Employes/Implementation-Java/Vendeur.java
class Vendeur extends Employe {
    private double commission;

    public Vendeur(String numeroSecuriteSociale, String nom, double salaire, Employe superieur, double commission) {
        super(numeroSecuriteSociale, nom, salaire, "Vendeur", superieur);
        this.commission = commission;
    }

    @Override
    public double calculerSalaire() {
        return salaire + commission;
    }

    @Override
    public String afficherSubordonnes() {
        return super.afficherSubordonnes() + " Aucuns";
    }
}

File: Semestre-1/Cours-POO/Examen-Final_2022-2023/Exercice-2_Gestion-Employes/Implementation-Cpp/Vendeur.cpp
#include "Vendeur.h"

Vendeur::Vendeur(string numeroSecuriteSociale, string nom, double salaire, Employe * superieur, double commission) : Employe(numeroSecuriteSociale, nom, salaire, "Vendeur", superieur) {
    this->commission = commission;
}

double Vendeur::calculerSalaire() const {
    return salaire + commission;
}

string Vendeur::afficherSubordonnes() const {
    return Employe::afficherSubordonnes() + " Aucun";
}